src/lib/support/TimerDelegates.h

```diff
@@ -0,0 +1,66 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/InteractionModelEngine.h>
+#include <app/reporting/ReportScheduler.h>
+#include <system/SystemClock.h>
+
+namespace chip {
+
+class DefaultTimerDelegate : public app::reporting::ReportScheduler::TimerDelegate
+{
+public:
+    using ReadHandlerNode        = app::reporting::ReportScheduler::ReadHandlerNode;
+    using InteractionModelEngine = app::InteractionModelEngine;
+    using Timeout                = System::Clock::Timeout;
+    static void TimerCallbackInterface(System::Layer * aLayer, void * aAppState)
+    {
+        ReadHandlerNode * node = static_cast<ReadHandlerNode *>(aAppState);
+        node->RunCallback();
+    }
+    CHIP_ERROR StartTimer(void * context, Timeout aTimeout) override
+    {
+        return InteractionModelEngine::GetInstance()->GetExchangeManager()->GetSessionManager()->SystemLayer()->StartTimer(
```

Comment:
You can't have src/lib/support depend on InteractionModelEngine or src/app or whatnot.  If this header were properly added to the gn sources, gn would catch this for you....

This thing needs to live in src/app.